enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,156 @@
+---
+title: openstack-baremetal-compute
+authors:
+  - "@pierreprinetti"
+reviewers:
+  - "@openshift/team-openstack-cloud-provider"
+approvers:
+  - TBD
+creation-date: 2019-12-10
+last-updated: 2019-12-10
+status: provisional
+
+---
+
+# Openstack Baremetal Compute
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenStack's [Nova][openstack-nova] service can be configured to provision
+baremetal machines as OpenShift Compute Nodes.
+
+This enhancement will allow [Ironic][openstack-ironic]-provisioned machines to
+be used as OpenShift Compute Nodes, when they are made available as Nova
+flavors.
+
+## Motivation
+
+In heavy-loaded clusters, or when the workload is better handled with specially
+purposed hardware, the functional requirement of controlling where the work is
+computed might outweight the added value of virtualisation.
+
+In OpenStack clusters where Nova transparently provisions baremetal machines,
+this enhancement will ensure that OpenShift installations can use them as
+Compute Nodes.
+
+### Goals
+
+* Enable running OpenShift on a mix of virtual and baremetal machines.
+* Baremetal machines can be added and removed from a running OpenShift cluster.
+
+### Non-Goals
+
+**Separate subnets.**
+
+An OpenStack setting where baremetal machines can't be attached to a virtual
+subnet is not supported in this enhancement.
+
+**External connectivity via floating IPs.**
+
+The OpenStack cluster might make the BM nodes accessible, but exposing them via
+floating IPs might represent an increase in complexity and is currently not a
+goal.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+* The choice between deploying a VM or BM node is done by selecting an appropriate Nova flavour
+  * which can be done both in install config and passed to Cluster API provider OpenStack
+  * for the time being, the interactive prompt does ask the flavor for the Compute Nodes specifically; the choice must be made editing `install-config.yaml`
```

Comment:
It's s little more complicated than that. The interactive prompt asks for the [`computeFlavor` variable](https://github.com/openshift/installer/blob/master/docs/user/openstack/customization.md#cluster-scoped-properties), however it sets the flavor used for both the compute _and_ control-plane nodes. You can set different flavors for the Compute and Control Plane nodes by setting the `type` property of the `compute` and `controlPlane` [machine-pools structs](https://github.com/openshift/installer/blob/master/docs/user/openstack/customization.md#machine-pools) in `install-config.yaml`.

So to summarize, by using the interactive prompt you'll get the same profile for Control Plane and Compute nodes and it would be all BM or all VM, while you can use different flavors (and mix BM and VM) by using `install-config.yaml`.

Reply:
that's a typo; there should have been a _not_